tableaudocumentapi/datasource.py

```diff
@@ -141,6 +178,16 @@ def fields(self):
         return self._fields
 
     def _get_all_fields(self):
-        column_objects = (_mapping_from_xml(self._datasourceTree, xml)
-                          for xml in self._datasourceTree.findall('.//column'))
-        return MultiLookupDict({k: v for k, v in column_objects})
+        column_objects = [_column_object_from_column_xml(self._datasourceTree, xml)
+                          for xml in self._datasourceTree.findall('.//column')]
+        existing_fields = [x[0] for x in column_objects]
```

Comment:
Is there a way to have column objects return the element we need so there isn't a magic [0] here?

If _column_object_from_column_xml is just used in this function it would be clearer to the reader